kickstarter/android-oss
# EP-241: Create event for Discover sort CTA

# 📲 What

- Added a track event for Discovery tab click events that fires when the sorting tabs are clicked on the Discovery screen.

# 🤔 Why

Segment integration.

# 🛠 How

- Created a new function to our `AnalyticsEvents` class that handles sending the event name and properties to the client.
- Added this function to the `DiscoveryViewHolder` that fires when any of the tabs at the top of the screen are tapped
- .
# 👀 See

This are the sorting tabs that fire the events: 
![Screenshot_1614288807](https://user-images.githubusercontent.com/19390326/109220929-5d079380-7787-11eb-9981-c01a6076b88c.png)

# 📋 QA

- Load the app
- Tap the "Popular" sort tab
- Tap all the sorting tabs at the top of the screen: "Magic", "Popular", "Newest", "Ending Soon"
- Both Segment (CTA Clicked) and DataLake (Explore Sort Clicked) should appear in the segment dashboard. Each tab sort CTA should have it's own event, with "context_cta" = "discovery_sort", and "context_location" = "discover_advanced". 
- Each event should specify the correct tab clicked with the "discover_sort" property as either "magic", "popular", "newest", or "ending_soon"
<img width="784" alt="Screen Shot 2021-02-25 at 4 06 08 PM" src="https://user-images.githubusercontent.com/19390326/109223230-54fc2380-7788-11eb-91bf-b72ed2201fa8.png">
<img width="574" alt="Screen Shot 2021-02-25 at 4 06 19 PM" src="https://user-images.githubusercontent.com/19390326/109223647-de135a80-7788-11eb-8690-071e0f948564.png">
<img width="470" alt="Screen Shot 2021-02-25 at 4 06 33 PM" src="https://user-images.githubusercontent.com/19390326/109223648-de135a80-7788-11eb-8f9c-5f19db66e61d.png">
<img width="474" alt="Screen Shot 2021-02-25 at 4 06 43 PM" src="https://user-images.githubusercontent.com/19390326/109223649-de135a80-7788-11eb-8081-de82dbb10014.png">
<img width="471" alt="Screen Shot 2021-02-25 at 4 06 50 PM" src="https://user-images.githubusercontent.com/19390326/109223650-deabf100-7788-11eb-85aa-d929118519ae.png">

# Story 📖

[EP-223: Android CTA Clicked (discover_sort)](https://kickstarter.atlassian.net/browse/EP-241)
